src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5105,12 +5149,26 @@ private static BoundExpression SkipReferenceConversions(BoundExpression possibly
             var leftState = this.State.Clone();
             LearnFromNonNullTest(leftOperand, ref leftState);
             LearnFromNullTest(leftOperand, ref this.State);
+
+            // If we are assigning the underlying value type to a nullable value type variable,
```

Comment:
I found this comment a little unclear. I think it should say that we do a special step to set the nullable variable's top level state within this block, then change the slot to later simulate an assignment from the RHS to the LHS's Value property, to update nullable states within the underlying value type.